src/components/LibraryItem.tsx

```diff
@@ -245,4 +246,18 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
             libraryContainer.raiseEvent("itemClicked", this.props.data.contextData);
         }
     }
+
+    onLibraryItemHoveredOn() {
+        let libraryContainer = this.props.libraryContainer;
+        if (this.props.data.childItems.length == 0) {
+            libraryContainer.raiseEvent("itemHoveredOn", this.props.data.contextData);
```

Comment:
I was thinking if we can define string constants on library containers to define all the possible event names so that clients can easily figure out which events are available and also we can get rid of all the string hard coding for event names and any potential errors could be caught at compile time.